streamer/src/nonblocking/quic.rs

```diff
@@ -1241,21 +1241,21 @@ pub mod test {
             // the stream -- expect it.
             let s2 = s2.err().unwrap();
             assert!(matches!(s2, quinn::ConnectionError::ApplicationClosed(_)));
-            return;
+        } else {
```

Comment:
How is this change functionally different from the current code?

Reply:
https://buildkite.com/organizations/solana-labs/analytics/suites/solana-1/tests/018771d1-3d7f-7074-8db3-23bace9b9208?branch=master -- indicates the tests might be more reliable now